src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.ActionStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final ActionStrategy operationStrategy;
+
+    public ShopServiceImpl(ActionStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            ActionHandler handler = operationStrategy.getHandler(transaction.getAction());
+            handler.apply(transaction.getFruit(), transaction.getValue());
+        }
+    }
+}
```

Comment:
Don't begin class or method implementation with an empty line. Please remove the empty line at the beginning of the class definition (line 8) and the method `process` (line 16).